en/autogen/README.md

```diff
@@ -0,0 +1,68 @@
+# Autogeneration
+
+MAVSDK is available in a number of different languages.
+These all share the same "core" MAVLink implementation (written in C++), but almost all of the API-specific code for other languages is *autogenerated* from API definition files (and language-specific templates).
+
+This approach means that we don't have to maintain a separate MAVLink implementation or API for each language.
+New features are implemented (just once) in C++, and are then automatically available in Python, Mava, etc.
+
+This topic provides an overview of the architecture and mechanisms.
+
+## Overview
+
+The common MAVLink implementation is written in C++ (it is part of MAVSDK-C++). 
+The `mavsdk_server` exposes the MAVSDK-C++ API over [gRPC](https://grpc.io/) to the language bindings.
+The API for all languages is autogenerated from a single common definition.
+
+![MAVSDK structure/architecture](../../assets/autogen/mavsdk_overview.png)
+
+The following parts are autogenerated:
+
+* MAVSDK-C++ API (i.e. header files)
+* Most of `mavsdk_server`
+* Most of the language bindings
+
+Those parts are maintained manually:
+
+* MAVSDK-C++ implementation (i.e. the MAVLink business logic)
+* New plugins need to be added to `mavsdk_server` (this may be automated in future)
+* The `System` wrapper in language bindings usually needs to be updated whenever a new plugin is added (this may be automated in future)
+
+The heart of the autogeneration system is the [MAVSDK-Proto](https://github.com/mavlink/mavsdk-proto) repository, which is described [below](#mechanisms).
+
+## Autogeneration Mechanisms {#mechanisms}
+
+The autogeneration pipeline is shown below:
+
+![Autogeneration pipeline](../../assets/autogen/autogen_overview.png)
```

Comment:
I would add the proto files to this diagram as an input

Reply:
Not sure what you mean. I'm omitting this for now.